lib/wserver/wserver.ml

```diff
@@ -258,97 +258,81 @@ let check_stopping () =
     flush stderr;
     exit 0)
 
+let client_connection tmout callback addr t =
+  let oc = Unix.out_channel_of_descr t in
+  connection_closed := false;
+  wserver_sock := t;
+  wserver_oc := oc;
+  Fun.protect ~finally:close_connection @@ fun () ->
+  treat_connection tmout callback addr t
+
 let accept_connection tmout max_clients callback s =
   let () = wait_available max_clients s in
   let t, addr = Unix.accept s in
   check_stopping ();
   Unix.setsockopt t Unix.SO_KEEPALIVE true;
-  if Sys.unix then (
-    if !no_fork then (
-      connection_closed := false;
-      wserver_sock := t;
-      wserver_oc := Unix.out_channel_of_descr t;
-      treat_connection tmout callback addr t;
-      close_connection ())
-    else
-      match try Some (Unix.fork ()) with _ -> None with
-      | Some 0 -> (
-          try
-            if max_clients = None && Unix.fork () <> 0 then exit 0;
```

Comment:
This line is really strange. It's a common pattern to fork twice a process in order to demonize it but
we should do it even if there is a limit on the number of clients.

I didn't touch it in this PR to keep it as atomic as possible.